traffic_ops/v4-client/profile.go

```diff
@@ -77,9 +77,13 @@ func (to *Session) GetParametersByProfileName(profileName string, header http.He
 }
 
 // GetProfiles returns all Profiles stored in Traffic Ops.
-func (to *Session) GetProfiles(header http.Header) ([]tc.Profile, toclientlib.ReqInf, error) {
+func (to *Session) GetProfiles(params url.Values, header http.Header) ([]tc.Profile, toclientlib.ReqInf, error) {
+	uri := APIProfiles
+	if params != nil {
```

Comment:
Instead of checking for `nil`, we should just check that the length is greater than 1, because right now if you call it like `session.GetProfiles(url.Values{}, nil)` it'll use the path `profiles?`. That shouldn't pose a functional problem, but could potentially be confusing in log messages - people might think the actual query string wasn't printed properly, or mistake the <kbd>?</kbd> for punctuation.

Reply:
Fixed it.